lib/appsignal/config.ex

```diff
@@ -100,6 +98,17 @@ defmodule Appsignal.Config do
     Application.fetch_env!(:appsignal, :config).active
   end
 
+  @doc """
+  Returns true if the configuration is valid. Configuration is considered
+  valid if there's an push API key set.
+  """
+  @spec valid?() :: boolean
+  def valid? do
+    Application.get_env(:appsignal, :config)[:push_api_key]
+    |> empty?
+    |> Kernel.not()
+  end
```

Comment:
Didn’t get to this review before it was merged, but here’s a comment I wrote. No action needed, just some ideas:

Another way to do this, which more specifically checks if the push API key is a nonempty string would be like this:

``` elixir
def valid? do
  do_valid?(Application.get_env(:appsignal, :config)[:push_api_key])
end

defp do_valid?(push_api_key) when is_binary(push_api_key)
  String.length(push_api_key) > 0
end

defp do_valid?(_push_api_key) do
  false
end
```

By finding the API key in `valid?/0` and passing it to `do_valid?/1`, this function uses a guard clause to see if it’s a string. If so, its length is checked to make sure it’s not empty. Otherwise, it falls through, matches the last clause, and returns false.

Reply:
I also noticed the Elixir integration doesn't check for non-empty strings that only contains whitespace characters, such as `" "`. I've created a PR to standardize this validation so that we can update the Elixir integration to match: https://github.com/appsignal/integration-guide/pull/83

Reply:
Copy, I'll work on this now.

Reply:
Covered here: https://github.com/appsignal/appsignal-elixir/pull/721